src/main/java/bio/terra/workspace/service/resource/controlled/mappings/ControlledResourceInheritanceMapping.java

```diff
@@ -0,0 +1,91 @@
+package bio.terra.workspace.service.resource.controlled.mappings;
+
+import bio.terra.workspace.service.iam.model.ControlledResourceIamRole;
+import bio.terra.workspace.service.iam.model.WsmIamRole;
+import bio.terra.workspace.service.resource.controlled.AccessScopeType;
+import bio.terra.workspace.service.resource.controlled.ManagedByType;
+import bio.terra.workspace.service.workspace.exceptions.InternalLogicException;
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.Multimap;
+
+/**
+ * Static mappings of workspace-level IAM roles to resource-level IAM roles.
+ *
+ * <p>This mapping represents which resource-level roles are granted to workspace members based on
+ * their workspace-level role. This mapping is slightly different for each category of controlled
+ * resource (user-shared, user-private, app-shared, app-private), so this class holds four separate
+ * maps.
+ *
+ * <p>For example, all workspace readers should be readers of a user-shared bucket in that
+ * workspace, but the same is not true for a user-private bucket. This is represented in the maps
+ * below, where WsmIamRole.READER maps to ControlledResourceIamRole.READER for shared resources but
+ * that entry is not present for private resources.
+ *
+ * <p>This maps workspace roles to the resource roles, which are used for granting cloud
+ * permissions. Sam also has a similar mapping configured for granting Sam permissions. If you
+ * change this map, also check that mapping:
+ * https://github.com/broadinstitute/sam/blob/develop/src/main/resources/reference.conf
+ */
+public class ControlledResourceInheritanceMapping {
```

Comment:
fair point - I picked up the "inherited" language from the hierarchical resource design in Sam, where policies on "parent" resources are inherited by "child" resources. Since most of the resource IAM inside WSM is based on that feature in Sam